wagtail/wagtailadmin/edit_handlers.py

```diff
@@ -586,15 +631,24 @@ def render_js_init(self):
         }))
 
 
-def InlinePanel(base_model, relation_name, panels=None, label='', help_text=''):
-    rel = getattr(base_model, relation_name).related
-    return type(str('_InlinePanel'), (BaseInlinePanel,), {
-        'relation_name': relation_name,
-        'related': rel,
-        'panels': panels,
-        'heading': label,
-        'help_text': help_text,  # TODO: can we pick this out of the foreign key definition as an alternative? (with a bit of help from the inlineformset object, as we do for label/heading)
-    })
+class InlinePanel(object):
+    def __init__(self, base_model, relation_name, panels=None, label='', help_text=''):
```

Comment:
Would be great if we could deprecate the ``base_model`` parameter here.

I imagine the code for it won't be very pretty (it would only be temporary though):
 - If the first arg is a class, get ``relation_name`` from second arg and raise a ``DeprecationWarning``
 - Otherwise, treat the first arg as the ``relation_name``